### DISCUSSION TITLE
trame-pvui usage

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#348] [Q&A]
Hi!
I tried to run trame-pvui:
~/ParaView-5.11.0-RC2-MPI-Linux-Python3.9-x86_64/bin/pvpython example_trame_apps/paraview_app/main.py

The application starts and I can see the geometry, but in the menu, the tabs called 'Info Panel' and 'File Browser' do not show anything. In the tab 'Server Browser' I can see some buttons (load,save,fetch server) but they do not do anything.
I see in the konsole that remote dir, local dir and selected node are updated, but no other messages.

When starting, I see that I am using trame@3 and vue2, not sure if this is the wrong setup.

**Comment by bigfooted** [comment]
By the way, I am looking into this because trame-pvui seems to have a more advanced file handler. I would like to be able to select a file on the client as well as on the server (from the user folder) and load it (as well as save a file on the server or download it).

There is also a file browser in core/21_file_browser.py but that one only starts on the server and not on the client it seems.

**Comment by jourdain** [comment]
So the trame-pvui is meant for async-paraview so I'm not surprised if the "paraview" version is not fully functional anymore. 

Regarding trame v3 and vue2. That is fine. You should make sure you call `server.client_type = "vue2"` or `server = get_server(client_type="vue2")` to ensure that it will still work down the road. 

Regarding your hope of loading files either locally (browser) or remotely (server). This is not supported by that component. Those terms are meant for ParaView where the "local"(web-server) and "remote" (cluster) mean something different. 

HTH,

Seb

**Reply by bigfooted** [reply]
OK, thanks for your help with all these basic questions from my side, I am still struggling with finding out what the best way is to do the file handling. 
So just to be clear: the example using tkinter (21_file_browser) can get the complete path of a local file, so I could use that for a local setup where the file is not copied, and VFileInput will always upload the file and I have to work with the copy?

**Reply by jourdain** [reply]
Yes the tk approach is good if things are local (desktop setup). The VFileInput triggers an upload that you will have to manage server side. Or you can use the pvui widget to browse the server side (where the python is running).

HTH
